diff --git a/dist/index.mjs b/dist/index.mjs
index db9a8436597cc862f63a1eb04b73d62c7386826f..f20dc0b2b11bc29bee0427eed389ba614304b10d 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -171,7 +171,7 @@ function SvelteKitPlugin(options, apiResolver) {
       viteConfig = config;
     },
     async generateBundle(_, bundle) {
-      if (viteConfig.build.ssr)
+      if (this.environment.config.consumer === "server")
         return;
       const api = apiResolver();
       if (!api)
@@ -186,19 +186,18 @@ function SvelteKitPlugin(options, apiResolver) {
       enforce: "pre",
       async handler() {
         const api = apiResolver();
-        if (!api || viteConfig.build.ssr)
+        if (!api || this.environment.config.consumer === "server")
           return;
         const assetsGenerator = await api.pwaAssetsGenerator();
         if (assetsGenerator)
           await assetsGenerator.generate();
       }
     },
-    closeBundle: {
-      sequential: true,
-      enforce: "pre",
+    buildApp: {
+      order: "post",
       async handler() {
         const api = apiResolver();
-        if (api && !api.disabled && viteConfig.build.ssr) {
+        if (api && !api.disabled) {
           const webManifest = options.manifestFilename ?? "manifest.webmanifest";
           let swName = options.filename ?? "sw.js";
           const outDir = options.outDir ?? `${viteConfig.root}/.svelte-kit/output`;
diff --git a/package.json b/package.json
index c6214fd1f130f9e8a258e17cbf0bb3c8af0ca089..5cde3cf918842e21df9349bcff2dda4c10460f91 100644
--- a/package.json
+++ b/package.json
@@ -41,7 +41,7 @@
     "node": ">=18.13"
   },
   "peerDependencies": {
-    "@sveltejs/kit": "^1.3.1 || ^2.0.1",
+    "@sveltejs/kit": "^1.3.1 || ^2.0.1 || ^3.0.0",
     "@vite-pwa/assets-generator": "^1.0.0"
   },
   "peerDependenciesMeta": {
